### PR TITLE
feat: [CDS-85873]: Cleanup CD_TRIGGER_CATALOG_API_ENABLED & CD_TRIGGER_V2 FF

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
### Summary

**CD_TRIGGER_CATALOG_API_ENABLED** FF was used to show trigger catalog data using BE API instead of UI hard-coded list.
Removed this FF and related hard-coded lists and tests also.

**CD_TRIGGER_V2** FF was used in the trigger catalog and as we started using catalog API this FF is also not needed anymore in UI. So removed that also.

#### Screenshots
**Before:** Trigger Catalog is rendered using hardcode data in UI.

https://github.com/harness/harness-core-ui/assets/110591046/c840d5d2-560c-4fa3-9a64-5364decb907b

**After:** Trigger Catalog is rendered using trigger catalog API data.

https://github.com/harness/harness-core-ui/assets/110591046/0c02288e-9607-4afe-9436-7bbac4f1fb37

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by pankajsingh-harness** [comment]
trigger messagecheck

**Comment by pankajsingh-harness** [comment]
trigger messagecheck

File: cypress/integration/70-pipeline/PipelineTriggers.spec.ts [hunk 1]
```diff
@@ -40,41 +43,6 @@ describe('Triggers for Pipeline', () => {
     }
   })
 
-  it('Triggers Drawer & List Assertion', () => {
```
Comment (pankajsingh-harness) [inline cypress/integration/70-pipeline/PipelineTriggers.spec.ts:43]: Removed this as this test is Jest test is already there.

**Review by sonaligoyal-harness — DISMISSED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Review by ishantmehtaharness — DISMISSED** [review]
Is this FF enabled globally?
Looks good otherwise

**Review by sonaligoyal-harness — DISMISSED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

**Comment by pankajsingh-harness** [comment]
@ishantmehtaharness `CD_TRIGGER_V2` FF is enabled globally. `CD_TRIGGER_CATALOG_API_ENABLED` FF was added as there was some mismatch between the API response and the hardcode triggers list that we maintain in UI. `CD_TRIGGER_CATALOG_API_ENABLED` is just to enable trigger catalog list from API instead of UI hard-coded list. 
There is no other functionality behind this FF.

Also as we are not enabling the trigger catalog list from API, `CD_TRIGGER_V2` is not needed in UI anyone as this was used in the trigger catalog list to show AMI and Azure artifact rigger in the list. 
We have got QA Sign-Off for Azure and AMI artifact trigger. So BE will also be removing `CD_TRIGGER_V2` FF.

**Review by sonaligoyal-harness — APPROVED** [review]

**Pull request merged**